src/cascadia/WindowsTerminal/AppHost.h

```diff
@@ -122,4 +126,32 @@ class AppHost
     winrt::event_token _GetWindowLayoutRequestedToken;
     winrt::event_token _WindowCreatedToken;
     winrt::event_token _WindowClosedToken;
+
+    // Event handlers to revoke in ~AppHost, before calling App.Close
+    winrt::Microsoft::Terminal::Remoting::WindowManager::BecameMonarch_revoker _BecameMonarchRevoker;
+    winrt::Microsoft::Terminal::Remoting::Peasant::ExecuteCommandlineRequested_revoker _peasantExecuteCommandlineRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::Peasant::SummonRequested_revoker _peasantSummonRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::Peasant::DisplayWindowIdRequested_revoker _peasantDisplayWindowIdRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::Peasant::QuitRequested_revoker _peasantQuitRequestedRevoker;
+    winrt::TerminalApp::AppLogic::CloseRequested_revoker _CloseRequestedRevoker;
+    winrt::TerminalApp::AppLogic::RequestedThemeChanged_revoker _RequestedThemeChangedRevoker;
+    winrt::TerminalApp::AppLogic::FullscreenChanged_revoker _FullscreenChangedRevoker;
+    winrt::TerminalApp::AppLogic::FocusModeChanged_revoker _FocusModeChangedRevoker;
+    winrt::TerminalApp::AppLogic::AlwaysOnTopChanged_revoker _AlwaysOnTopChangedRevoker;
+    winrt::TerminalApp::AppLogic::RaiseVisualBell_revoker _RaiseVisualBellRevoker;
+    winrt::TerminalApp::AppLogic::SystemMenuChangeRequested_revoker _SystemMenuChangeRequestedRevoker;
+    winrt::TerminalApp::AppLogic::ChangeMaximizeRequested_revoker _ChangeMaximizeRequestedRevoker;
+    winrt::TerminalApp::AppLogic::TitleChanged_revoker _TitleChangedRevoker;
+    winrt::TerminalApp::AppLogic::LastTabClosed_revoker _LastTabClosedRevoker;
+    winrt::TerminalApp::AppLogic::SetTaskbarProgress_revoker _SetTaskbarProgressRevoker;
+    winrt::TerminalApp::AppLogic::IdentifyWindowsRequested_revoker _IdentifyWindowsRequestedRevoker;
+    winrt::TerminalApp::AppLogic::RenameWindowRequested_revoker _RenameWindowRequestedRevoker;
+    winrt::TerminalApp::AppLogic::SettingsChanged_revoker _SettingsChangedRevoker;
+    winrt::TerminalApp::AppLogic::IsQuakeWindowChanged_revoker _IsQuakeWindowChangedRevoker;
+    winrt::TerminalApp::AppLogic::SummonWindowRequested_revoker _SummonWindowRequestedRevoker;
+    winrt::TerminalApp::AppLogic::OpenSystemMenu_revoker _OpenSystemMenuRevoker;
+    winrt::TerminalApp::AppLogic::QuitRequested_revoker _QuitRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::WindowManager::ShowNotificationIconRequested_revoker _ShowNotificationIconRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::WindowManager::HideNotificationIconRequested_revoker _HideNotificationIconRequestedRevoker;
+    winrt::Microsoft::Terminal::Remoting::WindowManager::QuitAllRequested_revoker _QuitAllRequestedRevoker;
 };

```

Comment:
Woof. This is going to be the next candidate to be put into the macros.

Reply:
I quite nearly had a `std::deque<>` of event revokers that I just pushed them all into, then pop'd and revoked, but alas, I'm not lucky enough to have them all share some common base class ;____;